Network/Socket/Types.hsc

```diff
@@ -135,6 +135,10 @@ invalidateSocket (Socket ref _) errorAction normalAction = do
 --
 --   Sending data to or receiving data from closed socket
 --   may lead to undefined behaviour.
+--
+--   If multiple threads use the same socket and one uses 'fdSocket' and
+--   the other use 'close', unexpected behavior may happen.
+--   For more information, please refer to the documentation of 'fdSocket'.
```

Comment:
The issue affects more than just `fdSocket` and more than just `close`.  In general it is semantically unsafe for two threads to use the same socket, data may be corrupted, or sent to or read from the wrong place.  The only exceptions I can think of are:
* A reader thread can coëxist with a writer thread
* Multiple threads can call `accept` to process incoming connections

So I think this sort of advice belongs at the top of the package, not just the close function.

Reply:
yes, that will be nice.